core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -132,11 +135,14 @@ public ProbeResult doApply(Plugin plugin, ProbeContext context) {
 
                     for (DiffEntry entry : entries) {
                         if (zonedDateTime.isAfter(plugin.getReleaseTimestamp())) {
```

Comment:
Here my commit messages get updated on iteration.

But the file name remains the same.

Good new is my fail test counts has reduced to 4 from 5.

Reply:
This if should be an exit condition on the loop of commits